wasmplugin/plugin.go

```diff
@@ -365,3 +393,61 @@ func logError(error ctypes.MatchedRule) {
 		proxywasm.LogDebug(msg)
 	}
 }
+
+type AddressInfo struct {
+	IP   string
+	port int
+}
+
+// Retrieves adddress properties from the proxy
+// Expected targets are "source" or "destination"
+// Envoy ref: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes#connection-attributes
+func RetrieveAddressInfo(target string) AddressInfo {
+	var targetAddress AddressInfo
+	srcAddressRaw, err := proxywasm.GetProperty([]string{target, "address"})
+	if err != nil {
+		proxywasm.LogWarnf("failed to get %s address: %v", target, err)
+	} else {
+		targetAddress.IP, err = parseAddress(srcAddressRaw)
+		if err != nil {
+			proxywasm.LogWarnf("failed to parse %s address: %v", target, err)
+		}
+	}
+
+	srcPortRaw, err := proxywasm.GetProperty([]string{target, "port"})
+	if err != nil {
+		proxywasm.LogInfof("failed to get %s port: %v", target, err)
+	} else {
+		targetAddress.port, err = parsePort(srcPortRaw)
+		if err != nil {
+			proxywasm.LogWarnf("failed to parse %s port: %v", target, err)
+		}
+	}
+	return targetAddress
+}
+
+// Parses address (e.g. "127.0.0.1:8080") to retrieve the IP address
+func parseAddress(rawAddress []byte) (string, error) {
```

Comment:
Is it possible to use https://pkg.go.dev/net#SplitHostPort?

Reply:
Working with strings, definitely yes, thanks. Added a commit with it.